### PR TITLE
Fix error when commit occurs in topmost transaction with TAC enabled

### LINKED ISSUE #51: Error after updating to 1.2.0
I just updated to v1.2.0 and I'm getting this error when I seed the database:
```
NoMethodError: undefined method `records' for #<ActiveRecord::ConnectionAdapters::NullTransaction:0x000000000a51c048>
/app/heroku/ruby/bundle/ruby/2.4.4/gems/test_after_commit-1.2.0/lib/test_after_commit/database_statements.rb:45:in `test_commit_records'
/app/heroku/ruby/bundle/ruby/2.4.4/gems/test_after_commit-1.2.0/lib/test_after_commit/database_statements.rb:29:in `transaction'
```

Any hints on why this is happening?

### PR DESCRIPTION
**Pull request by printercu**
Fixes #51

### CODE REVIEW

**Comment by printercu** [comment]
omg. finally green. got some troubles with 3.2.

File: lib/test_after_commit/database_statements.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module TestAfterCommit::DatabaseStatements
   def transaction(*)
     @test_open_transactions ||= 0
+    skip_emulation = ActiveRecord::Base.connection.open_transactions.zero?
```
Comment (grosser) [inline lib/test_after_commit/database_statements.rb]: that looks like it was fun to fund out :D

**Comment by grosser** [comment]
thx! looks like a tiny change that took lot's of work :D

**Comment by grosser** [comment]
v1.2.1 🎉

**Pull request merged**